app-android/src/main/java/io/github/droidkaigi/confsched2023/CompleteAchievementActivity.kt

```diff
@@ -0,0 +1,38 @@
+package io.github.droidkaigi.confsched2023
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.ui.Modifier
+import dagger.hilt.android.AndroidEntryPoint
+import io.github.droidkaigi.confsched2023.animation.AnimationScreen
+import io.github.droidkaigi.confsched2023.designsystem.theme.KaigiTheme
+
+@AndroidEntryPoint
+class CompleteAchievementActivity : ComponentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            KaigiTheme {
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = MaterialTheme.colorScheme.background,
+                ) {
+                    AnimationScreen(
+                        deepLink = intent.data.toString(),
+                        onFinished = {
+                            val intent = Intent(this, MainActivity::class.java)
+                            startActivity(intent)
```

Comment:
Please consider the desired transitions after the animation finishes.

Currently, CompleteAchievementActivity will be visible when a user presses a back key on MainActivity. And then, if a user rotates their device, the animation restarts and show up MainActivity again.

```
MainActivity <- visible
CompleteAchievementActivity

// back key is pressed

CompleteAchievementActivity <- visible

// rotate their device
MainActivity <- visible
CompleteAchievementActivity
```

We have two options.

1. Just finish CompleteAchievementActivity right after the animation finishes.
2. Change the activity stack in the first place.

`1.` is so easy.

```
MainActivity <- visible
```

For `2`, we can call `startActivities(MainActivity's intent, CompleteAchievementActivity's intent` in ResolveDynamicLinksActivity, and allow CompleteAchievementActivity not to launch MainActivity and not to finish. The final state of activity stacks will be like the following:

```
CompleteAchievementActivity <- visible
MainActivity
```

Reply:
Am I correct in recognizing that finishAndRemoveTask() is called after reading MainActivity?🤔
I actually tried pressing the back key after reading NFC and transitioning to MainActivity and the stack was cleared.

Reply:
Fixed in this commit.🙏

0ce3e60bc3398901b5e2b409c3fe40e9ebb32fd3